proof_system/src/sub_protocols/ps_signature.rs

```diff
@@ -104,19 +118,12 @@ impl<'a, E: Pairing> PSSignaturePoK<'a, E> {
             .map(|(idx, _)| (*idx, CommitMessage::RevealMessage));
 
         let mut invalid_message_idx = None;
-        let all_messages = messages_to_commit_with_blindings
-            .merge_by(revealed_messages, |(a, _), (b, _)| a <= b)
-            .scan(None, |last, (idx, msg)| {
-                if last.map_or_else(|| idx != 0, |last| last + 1 != idx) {
-                    invalid_message_idx.replace(idx);
-
-                    None
-                } else {
-                    last.replace(idx);
-
-                    Some(msg)
-                }
-            });
+        let all_messages = take_while_pairs_satisfy(
```

Comment:
Nitpick: Logic to create `all_messages` and `messages_to_commit_with_blindings` is same as in `bbs_plus.rs` (in the same folder) except that enum used to express them slightly different.